lib/bolt/transport/sudoable/connection.rb

```diff
@@ -70,6 +70,33 @@ def execute(*_args)
           message = "#{self.class.name} must implement #{method} to execute commands"
           raise NotImplementedError, message
         end
+
+        def prepend_sudo_success(sudo_id, command_str)
+          "sh -c 'echo #{sudo_id} 1>&2; #{command_str}'"
+        end
+
+        # A helper to build up a single string that contains all of the options
+        # for privilage escalation.
+        def build_sudoable_command_str(command_str, sudo_str, sudo_id, options)
+          if options[:stdin] && !options[:wrapper]
```

Comment:
This could definitely use a comment explaining why it works this way.

Reply:
Added some more detail.